src/Octoliner.cpp

```diff
@@ -120,7 +122,50 @@ uint8_t Octoliner::digitalReadAll(void) {
     return mapAnalogToPattern(analogValues);
 }
 
-void Octoliner::changeAddr(uint8_t newAddr) { 
+bool Octoliner::optimizeSensitivityOnBlackField(void) {
+    uint8_t count = 0;
+    uint8_t sens;
+    // backward quick search then all sensors value will be more then 100
+    for (sens = 255; sens > 0; sens -= 5) {
+        setSensitivity(sens);
+        delay(100);
+        count = 0;
+        for (uint8_t i = 0; i < 8; i++) {
+            if (analogRead(i) > 100)
```

Comment:
```suggestion
            if (analogRead(i) > BLACK_THRESHOLD)
```

Extract the value to global:

```cpp
// All analog values above this are considered black
constexpr int16_t BLACK_THRESHOLD = 100;
```